File: apps/api/app/schemas/restoration.py
from enum import Enum

from pydantic import BaseModel


class RestorationMode(str, Enum):
    CONSERVATIVE = "conservative"
    BALANCED = "balanced"
    STRONG = "strong"


class RestorationResponse(BaseModel):
    id: str
    original_file_name: str
    input_url: str
    output_url: str
    provider: str
    fallback_used: bool
    ai_model: str | None
    provider_error_code: str | None
    provider_error_message: str | None
    mode: RestorationMode
    message: str

File: apps/web/types/restoration.ts
export type RestorationMode = "conservative" | "balanced" | "strong";

export interface RestorationResponse {
  id: string;
  original_file_name: string;
  input_url: string;
  output_url: string;
  provider: "deterministic" | "openai" | string;
  fallback_used: boolean;
  ai_model: string | null;
  provider_error_code: string | null;
  provider_error_message: string | null;
  mode: RestorationMode;
  message: string;
}

export interface RestoreImageInput {
  file: File;
  mode: RestorationMode;
  useAi: boolean;
  onUploadProgress?: (progress: number) => void;
}
